File: main/encryption/encryption.rb
# determine dimensions for "ciphertext" grid
s = gets.gsub(/\s+/, '').freeze
n = s.size
sqrt = Math.sqrt(n)
h = sqrt.floor.to_int
w = sqrt.ceil.to_int
h += 1 if h * w < n

# create empty grid and populate it with ciphertext
newcol = (' ' * h).freeze
grid = (1..w).map { newcol.dup } # column-major ciphertext representation
(0...n).each { |k| grid[k % w][k / w] = s[k] }

# output ciphertext
puts grid.each(&:rstrip!).join(' ')

File: main/encryption/encryption.cs
using System;

internal static class Solution {
    private static void Main()
    {
        var s = Console.ReadLine().Replace(" ", string.Empty);
        var n = s.Length;
        var sqrt = Math.Sqrt(n);
        var h = Convert.ToInt32(Math.Floor(sqrt));
        var w = Convert.ToInt32(Math.Ceiling(sqrt));
        if (h * w < n) ++h;

        var grid = new char[w][]; // column-major ciphertext representation
        var newcol = new string(' ', h);
        for (var j = 0; j < w; ++j)
            grid[j] = newcol.ToCharArray();

        for (var k = 0; k < n; ++k)
            grid[k % w][k / w] = s[k];

        var words = Array.ConvertAll(grid, col => new string(col).TrimEnd());
        Console.WriteLine(string.Join(" ", words));
    }
}
